src/Forms/Components/MoneyInput.php

```diff
@@ -35,13 +35,14 @@ protected function setUp(): void
             return MoneyFormatter::formatAsDecimal($state, $currency, $locale);
         });
 
-        $this->dehydrateStateUsing(function (MoneyInput $component, $state): string {
-
+        $this->dehydrateStateUsing(function (MoneyInput $component, $state): ?string {
             $currency = $component->getCurrency();
             $state = MoneyFormatter::parseDecimal($state, $currency, $component->getLocale());
 
-            $this->prepare();
-
+            if (empty($state)) {
```

Comment:
I don't think we can use `empty` here. This will for example trigger when the value is 0. Maybe an `! is_numeric` check would do the job?

Reply:
I think is nummeric or $state == '' will do!

Can you fix it or shall i fix it?

Reply:
I can do it

Reply:
Thanks for fixing it!